api/v1/pooler_types.go

```diff
@@ -72,6 +72,10 @@ type PoolerSpec struct {
 
 	// The deployment strategy to use for pgbouncer to replace existing pods with new ones
 	DeploymentStrategy *appsv1.DeploymentStrategy `json:"deploymentStrategy,omitempty"`
+
+	// Enable or disable the `PodMonitor`
+	// +kubebuilder:default:=false
+	EnablePodMonitor bool `json:"enablePodMonitor,omitempty"`
```

Comment:
What about moving this in a dedicated MonitoringConfiguration? This way we would mirror what we are already doing in the Cluster resource, and if down the line we want to add custom queries we can do so without breaking the APIs.